File: src/main/java/com/projecttemplategroup/projecttemplateartifact/controllers/SignInController.java
package com.projecttemplategroup.projecttemplateartifact.controllers;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

//todo move to scala
@RestController
public class SignInController extends SecurityController{

    @GetMapping("/v1/signin")
    public String signIn(HttpServletRequest request, HttpServletResponse response) {
        return getSignedInProfile(request,response).toString();
    }
}

File: src/main/scala/com/projecttemplategroup/projecttemplateartifact/controllers/SignInController.scala
package com.projecttemplategroup.projecttemplateartifact.controllers

import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
import org.springframework.web.bind.annotation.{GetMapping, RestController}

@RestController
class SignInController extends SecurityController {

  @GetMapping(Array("/v1/signin"))
  def signIn(implicit request: HttpServletRequest, response: HttpServletResponse): String = {
    getSignedInProfile.toString
  }

}
